src/quo2/components/record_audio/record_audio/view.cljs

```diff
@@ -602,8 +676,91 @@
           :accessibility-label :mic-button}
          [icons/icon :i/audio {:color (colors/theme-colors colors/neutral-100 colors/white)}]]]))])
 
-(defn input-view
-  []
+(defn- soundtrack
+  [recording-length-ms ready-to-delete?]
+  [:f>
+   (fn []
+     (let [fill-percentage (/ (* recording-length-ms 100) max-audio-duration-ms)]
+       [rn/view {:style (style/soundtrack-container)}
+        [rn/view {:style (style/soundtrack-bar fill-percentage ready-to-delete?)}]]))])
+
+(defn- player-slider
+  [audio-current-time-ms player-ref seeking-audio?]
+  [:f>
+   (fn []
+     (let [audio-duration-ms (audio/get-player-duration @player-ref)]
+       [:<>
+        [rn/slider
+         {:style                    (style/player-slider-container)
+          :minimum-value            0
+          :maximum-value            audio-duration-ms
+          :value                    @audio-current-time-ms
+          :on-sliding-start         #(reset! seeking-audio? true)
+          :on-sliding-complete      (fn [seek-time]
+                                      (reset! seeking-audio? false)
+                                      (audio/seek-player
+                                       @player-ref
+                                       seek-time
+                                       #(log/debug "[record-audio] on seek - seek time: " seek-time)
```

Comment:
Not something for your PR, but handling errors simply with logs doesn't lead to a good user experience. If a failure to seek happens, users should get UI feedback in some way.

Another thing, I see that a lot of the designs do not cater for failure. Lots of stuff are just assumed to always work, but we know how this story goes in production. I worked in many PRs where all I could do was log, but this is hardly a solution IMO.

Reply:
I think we (the engineering team) should start to bring this topic to designers in each and every work that we do, otherwise we'll continue to accumulate unrealistic designs that only work for the happy path.

Reply:
In this particular case I feel it would have been difficult to detect this possible unhappy path because I wasn't aware of the audio player API. In fact, I've never experienced the on-error callback to be called using this function, but I agree we need to be more cautious and detect as many unhappy paths as we can at an early stage, and request designs for them.

Reply:
>In fact, I've never experienced the on-error callback to be called using this function

We have lots (and I mean lots) of error conditions that pretty much never happen, but if/when they do, the user will be out of luck and be extremely frustrated. For example, RPC calls to status-go should almost never fail, so what do we do? We just log, which is pointless to the user. Okay, maybe useful to send reports, but I remember the QA team saying they were pretty much useless.

I can totally see that once we have real users, the team's and org's mindset will change about how errors should be treated/designed.

>but I agree we need to be more cautious and detect as many unhappy paths as we can at an early stage, and request designs for them.

:+1: I'll keep that in mind too.